core/storage/learner_group/gae_models.py

```diff
@@ -0,0 +1,247 @@
+        """
+        if cls.get_by_id(group_id):
+            raise Exception(
+                'A learner group with the given group ID exists already.')
+
+        entity = cls(id=group_id, title=title, description=description)
+
+        entity.update_timestamps()
+        entity.put()
+        return entity
+
+    @classmethod
+    def export_data(cls, user_id: str) -> Dict[str, Dict[str, List[str]]]:
+        """Takeout: Export LearnerGroupModel user-based properties.
+
+        Args:
+            user_id: str. The user_id denotes which user's data to extract.
+
+        Returns:
+            dict. A dict containing the user-relevant properties of
+            LearnerGroupModel.
+        """
+        found_models = cls.get_all().filter(
+            datastore_services.any_of(
+                cls.student_user_ids == user_id,
+                cls.invited_user_ids == user_id,
+                cls.facilitator_user_ids == user_id
+        ))
+        user_data = {}
+        for learner_group_model in found_models:
+            user_data[learner_group_model.id] = {
+                'title': learner_group_model.title,
+                'description': learner_group_model.description,
+                'subtopic_page_ids':
+                    learner_group_model.subtopic_page_ids,
+                'story_ids': learner_group_model.story_ids
+            }
+        return user_data
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id: str) -> bool:
+        """Check whether LearnerGroupModel contains data of a given user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return (
+            cls.query(datastore_services.any_of(
+                cls.student_user_ids == user_id,
+                cls.invited_user_ids == user_id,
+                cls.facilitator_user_ids == user_id
+            )).get(keys_only=True) is not None
+        )
+
+    @classmethod
+    def apply_deletion_policy(cls, user_id: str) -> None:
+        """Delete all LearnerGroupModel instances associated with the
+        user.
+
+        Args:
+            user_id: str. The user_id denotes which user's data to delete.
+        """
+        found_models = cls.get_all().filter(
+            datastore_services.any_of(
+                cls.student_user_ids == user_id,
+                cls.invited_user_ids == user_id,
+                cls.facilitator_user_ids == user_id
+        ))
+
+        for learner_group_model in found_models:
+            # If the user is the facilitator of the group and there is
+            # only one facilitator_user_id, delete the group.
+            if user_id in learner_group_model.facilitator_user_ids and (
+                    len(learner_group_model.facilitator_user_ids) == 1):
+                # Remove references of the group from all related learner
```

Comment:
If you just drop this then how will the affected students know what happened to their group when they next check the website?

Reply:
We can add the notifications thing here when we implement it, but for now, it would show page not found to them. We talked about implementing this later in the green light doc.